cmd/operator-sdk/generate/csv.go

```diff
@@ -96,61 +121,49 @@ Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config
 }
 
 func (c csvCmd) run() error {
-
-	absProjectPath := projutil.MustGetwd()
-	cfg := &input.Config{
-		AbsProjectPath: absProjectPath,
-		ProjectName:    filepath.Base(absProjectPath),
-	}
-	if projutil.IsOperatorGo() {
-		cfg.Repo = projutil.GetGoPkg()
-	}
-
 	log.Infof("Generating CSV manifest version %s", c.csvVersion)
 
-	csvCfg, err := catalog.GetCSVConfig(c.csvConfigPath)
-	if err != nil {
-		return err
-	}
 	if c.operatorName == "" {
-		// Use config operator name if not set by CLI, i.e. prefer CLI value over
-		// config value.
-		if c.operatorName = csvCfg.OperatorName; c.operatorName == "" {
-			// Default to using project name if both are empty.
-			c.operatorName = filepath.Base(absProjectPath)
-		}
-	}
-
-	s := &scaffold.Scaffold{}
-	csv := &catalog.CSV{
-		CSVVersion:     c.csvVersion,
-		FromVersion:    c.fromVersion,
-		ConfigFilePath: c.csvConfigPath,
-		OperatorName:   c.operatorName,
+		c.operatorName = filepath.Base(projutil.MustGetwd())
 	}
-	err = s.Execute(cfg, csv)
-	if err != nil {
-		return fmt.Errorf("catalog scaffold failed: %v", err)
+	cfg := gen.Config{
+		OperatorName: c.operatorName,
+		// TODO(hasbro17): Remove the Input key map when the Generator input keys
+		// are removed in favour of config fields in the csvGenerator
```

Comment:
The comment here is not very clear for me. What is the requirement for we are able to remove it? Can we link an issue to know what is expected to get done first? Could we clarify more this? WDYT?

Reply:
> Can we link an issue to know what is expected to get done first?

@camilamacedo86 This is the follow up work that needs to happen after this PR gets merged. @estroz and I discussed this above:
https://github.com/operator-framework/operator-sdk/pull/2511#discussion_r378385371

As mentioned above I will file the actual issue(s) after this is merged because it only becomes an issue after this is merged.

**In summary:** We will replace the `Inputs` key value map with properly defined fields in the CSV and CRD generator structs.
The reason is to to reduce the complexity of passing inputs to the CSV generator so it's easier to understand and maintain.

Quite similar to how we replaced the key value inputs flag `inputs=<key1>=<val1>,<key2>=<val2>` flag with defined flags like `--deploy-dir=deploy`, `--apis-dir=pkg/apis` etc. I mentioned this in response to your earlier review comments as well.
https://github.com/operator-framework/operator-sdk/pull/2511#discussion_r380383154

Again I'm keeping it out of this PR to keep the scope minimal for something that will be a purely internal change with no user facing or CLI changes.